modules/vba_documents/app/sidekiq/vba_documents/upload_processor.rb

```diff
@@ -94,17 +97,32 @@ def original_file_metadata(tempfile)
       }
     end
 
-    def find_icn(file_number)
-      bgss = BGS::Services.new(external_uid: file_number, external_key: file_number)
+    def read_original_metadata_file_number(parts)
+      return unless parts.key?(META_PART_NAME) && parts[META_PART_NAME].is_a?(String)
+
+      metadata = JSON.parse(parts[META_PART_NAME])
+      return unless metadata.is_a?(Hash) && metadata.key?('fileNumber')
+
+      return if (FILE_NUMBER_REGEX =~ metadata['fileNumber'].strip).nil?
+
+      metadata['fileNumber'].strip
+    end
+
+    def find_icn(parts)
+      consumer_file_number = read_original_metadata_file_number(parts)
 
```

Comment:
I added this little helper method to attempt to just go after the file_number field in the metadata that the consumer sends us.

UploadValidator is called below and also goes after the fileNumber field and validates it's presence and format, however, it's all grouped up with validating all the other metadata fields, so for example, if a metadata field other than fileNumber were missing, UploadValidator would throw an exception and i'd never get the fileNumber field back even if it were valid.  I want to record icn regardless of the state of the other metadata fields the consumer provided

Reply:
Does this mean that we have the code duplicated (one implementation here and the other in the validations)?

I'm wondering how important it really is to store the ICN if the submission fails our internal validations. But if we need to, I think this should be approached differently so that we don't need to maintain multiple implementations of the same code. The code should be placed in a shared location.

Reply:
Confirmed my assumption with PO, he wants us to always make a best effort to go after that ICN.

My read_original_metadata_file_number uses UploadValidator's fileNumber regex, but other than that they are different, UploadValidator throws exceptions when it encounters issues.....my read_original_metadata_file_number does not...it's more of a safe read about as simple as  I could make it, no logging, exceptions or anything like that.

Reply:
^For to mention, i did move read_original_metadata_file_number to the UploadValidations module, i was on the fence, but your comment kicked it over for me